### PR TITLE
Change scope

### PR DESCRIPTION
**Pull request by da-steve101**
This pull request makes a few functions private[Chisel] to clean up the API
Also makes Module extend nameable rather than redeclaring name and named
Bits should return a Bits object rather than a UInt and fix cases where this caused errors
Changed toBits : UInt to toBits : Bits, Added toUInt : UInt to replace toBits : UInt
Deprecate DC in Bool in favour of BitPat, ROM needs fixing for deprecation

Could cause some errors if creating Bits rather than UInt as does not have trait Num

### CODE REVIEW

File: src/main/scala/Module.scala [hunk 1]
```diff
@@ -278,8 +287,8 @@ abstract class Module(var _clock: Option[Clock] = None, private[Chisel] var _res
   def <>(src: Module) { io <> src.io }
 
   def apply(name: String): Data = io(name)
-  // COMPILATION OF REFERENCE, TODO: make private[Chisel]?
-  def emitDec(b: Backend): String = {
+  // COMPILATION OF REFERENCE
+  private[Chisel] def emitDec(b: Backend): String = {
```
Comment (donggyukim) [inline src/main/scala/Module.scala]: This is deprecated, so we can remove it

File: src/main/scala/Module.scala [hunk 2]
```diff
@@ -307,9 +316,8 @@ abstract class Module(var _clock: Option[Clock] = None, private[Chisel] var _res
     gen
   }
 
-  // TODO: should be private[Chisel]?
   /** Add a submodule to this module */
-  def addModule[T<:Module](c: =>T, f: PartialFunction[Any,Any]) = {
+  private[Chisel] def addModule[T<:Module](c: =>T, f: PartialFunction[Any,Any]) = {
```
Comment (donggyukim) [inline src/main/scala/Module.scala]: This is used in custom transforms, so should be public

File: src/main/scala/Module.scala [hunk 3]
```diff
@@ -320,9 +328,9 @@ abstract class Module(var _clock: Option[Clock] = None, private[Chisel] var _res
     Driver.compStack.pop
     res
   }
-  // TODO: should be private[Chisel]?
+
   /** Add a submodule to this module */
-  def addModule[T <: Module](c: => T)(implicit p:Parameters = params) = {
+  private[Chisel] def addModule[T <: Module](c: => T)(implicit p:Parameters = params) = {
```
Comment (donggyukim) [inline src/main/scala/Module.scala]: This is used in custom transforms, so should be public

File: src/main/scala/Module.scala [hunk 4]
```diff
@@ -334,9 +342,8 @@ abstract class Module(var _clock: Option[Clock] = None, private[Chisel] var _res
     res
   }
 
-  // TODO: should be private[Chisel]?
   /** A breadth first search of the graph of nodes */
-  def bfs (visit: Node => Unit) = {
+  private[Chisel] def bfs (visit: Node => Unit) = {
```
Comment (donggyukim) [inline src/main/scala/Module.scala]: This is used in custom transforms, so should be public

File: src/main/scala/Module.scala [hunk 5]
```diff
@@ -366,9 +373,8 @@ abstract class Module(var _clock: Option[Clock] = None, private[Chisel] var _res
     }
   }
 
-  // TODO: should be private[Chisel]?
   /** A depth first search of the graph of nodes */
-  def dfs(visit: Node => Unit): Unit = {
+  private[Chisel] def dfs(visit: Node => Unit): Unit = {
```
Comment (donggyukim) [inline src/main/scala/Module.scala]: This is used in custom transforms, so should be public

**Comment by ucbjrl** [comment]
Could we have this resubmitted removing the toBits() changes (or split into two pr's)? Bits is an abstract class; creating Bits objects is problematic. I don't think there should even be a toBits() method. In any case, changing toBits() to return a Bits instead of a UInt has far-reaching implications and will break a considerable number of existing designs.

**Comment by da-steve101** [comment]
I agree. Resubmitting to deprecate rather than breaking anything. Can be something down the track to change

**Comment by da-steve101** [comment]
Just submitted pr #504 with only scope and nameable changes
Having an issue with deprecating toBits in favour of toUInt as Flo overrides toUInt to convert to a unsigned int
changing toBits to toUInt means that Flo will return a conversion to int rather than just the bits as desired
Could change to asUInt but this breaks convention.
Ill submit the other pull request deprecating the Bits factory methods but i wont touch the toBits/toUInt

**Comment by da-steve101** [comment]
submitted  pr #506 with changes above

**Pull request closed without merging**